test/controllers/donation_goal_controller_test.exs

```diff
@@ -0,0 +1,108 @@
+defmodule CodeCorps.DonationGoalControllerTest do
+  use CodeCorps.ApiCase, resource_name: :donation_goal
+
+  alias CodeCorps.DonationGoal
+
+  @valid_attrs %{title: "A donation", description: "A description", amount: 200, current: false}
+  @invalid_attrs %{title: nil, description: nil}
+
+  describe "index" do
+    test "lists all entries on index", %{conn: conn} do
+      path = conn |> path_for(:index)
+      json = conn |> get(path) |> json_response(200)
+
+      assert json["data"] == []
+    end
+  end
+
+  describe "show" do
+    test "shows chosen resource", %{conn: conn} do
+      donation_goal = insert(:donation_goal)
+      json = conn |> request_show(donation_goal) |> json_response(200)
+
+      assert json["data"]["id"] |> String.to_integer == donation_goal.id
```

Comment:
Thoughts on leveraging the `assert_result_id` helper we have?

```
conn
|> request_show(donation_goal)
|> json_response(200)
|> Map.get("data")
|> assert_result_id(donation_goal.id)
```

Then we don't even have to assign `json`